pynoc/cisco.py

```diff
@@ -330,45 +296,35 @@ def _shorthand_port_notation(self, port):
 
         return output
 
-    def _send_commands(self, commands):
-        """Send a list of commands to the SSH shell.
-
-        :param commands: list of commands
-        """
-        for command in commands:
-            self._send_command(command[0], command[1])
-
-    def _send_command(self, command, signals, log=True):
-        """Send a command to the SSH shell.
+    def _send_command(self, command):
+        """Send command.
 
-        Sends a command to the SSH shell and waits for the signals to arrive.
+        Sends a command to the switch.
 
         :param command: command to send
-        :param signals: signals to wait for
-        :param log: specifies if this command should be logged
         :return: output of the command
         """
-        send_command = command + '\n'
-        if log:
-            self._logger.info('Sending command: %s', send_command.strip())
+        read_buffer = ""
+        if self._client is None:
+            return read_buffer
 
-        self._shell.send(send_command)
+        self._client.clear_buffer()
+        read_buffer = self._client.send_command(command)
```

Comment:
Nice. This is so much cleaner. I think this means we can remove the dependency on `six`.